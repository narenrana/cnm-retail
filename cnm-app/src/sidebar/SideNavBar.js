import React from 'react';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Collapse from '@material-ui/core/Collapse';
import ExpandLess from '@material-ui/icons/ExpandLess';
import ExpandMore from '@material-ui/icons/ExpandMore';
import StarBorder from '@material-ui/icons/StarBorder';
import Divider from '@material-ui/core/Divider';
import options from './data';
import useStyles from './style';

export default function SideBarList() {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);

  const handleClick = () => {
    setOpen(!open);
  };

  if (options) {
    return (
      <div className={classes.root}>
        <List>
          {options.map((option, index) =>
            <React.Fragment key={'list-button' + option.label + index}>
              <ListItem button onClick={handleClick} >
                <ListItemText primary={option.label} />
                {option.children && (open ? <ExpandLess /> : <ExpandMore />)}
              </ListItem>

              {option.children && <Collapse in={open} timeout="auto" unmountOnExit>
                <List component="div" disablePadding>
                  {option.children.map(child => <ListItem button className={classes.nested} key={'list-item' + (child.label) + index}>
                    <ListItemIcon>
                      <StarBorder />
                    </ListItemIcon>
                    <ListItemText primary={child.label} />
                  </ListItem>)}
                </List>
              </Collapse>}
              <Divider />
            </React.Fragment>
          )}
        </List>
      </div>
    )
  }
}
