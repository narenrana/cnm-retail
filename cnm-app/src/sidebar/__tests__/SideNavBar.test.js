
// Auto-generated do not edit


/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import SideNavBar from '../SideNavBar';


describe('SideNavBar test', () => {
  it('SideNavBar should match snapshot', () => {
    const component = renderer.create(<SideNavBar
       />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
