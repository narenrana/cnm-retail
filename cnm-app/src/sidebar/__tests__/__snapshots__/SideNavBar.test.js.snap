// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavBar test SideNavBar should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Fruits
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Organic
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          New Arrival
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Produce
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Meat
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Seafood
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Meal Kites
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Deli
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Dairy
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Beverages
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Bread
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Frozen
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Rice
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
  </ul>
</div>
`;
